lib/utils/platform/iot_threads_afr.c

```diff
@@ -0,0 +1,350 @@
+#include "platform/iot_threads.h"
+#include "types/iot_platform_types.h"
+
+/* Configure logs for the functions in this file. */
+#ifdef IOT_LOG_LEVEL_PLATFORM
+    #define _LIBRARY_LOG_LEVEL        IOT_LOG_LEVEL_PLATFORM
+#else
+    #ifdef IOT_LOG_LEVEL_GLOBAL
+        #define _LIBRARY_LOG_LEVEL    IOT_LOG_LEVEL_GLOBAL
+    #else
+        #define _LIBRARY_LOG_LEVEL    IOT_LOG_NONE
+    #endif
+#endif
+
+#define _LIBRARY_LOG_NAME    ( "THREAD" )
+#include "iot_logging_setup.h"
+
+/*
+ * Provide default values for undefined memory allocation functions based on
+ * the usage of dynamic memory allocation.
+ */
+#ifndef IotThreads_Malloc
+    #include <stdlib.h>
+
+    /**
+     * @brief Memory allocation. This function should have the same signature
+     * as [malloc](http://pubs.opengroup.org/onlinepubs/9699919799/functions/malloc.html).
+     */
+    #define IotThreads_Malloc    malloc
+#endif
+#ifndef IotThreads_Free
+    #include <stdlib.h>
+
+    /**
+     * @brief Free memory. This function should have the same signature as
+     * [free](http://pubs.opengroup.org/onlinepubs/9699919799/functions/free.html).
+     */
+    #define IotThreads_Free    free
+#endif
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @defgroup Mutex types.
+ */
+/**@{ */
+#ifndef IOT_THREAD_MUTEX_NORMAL
+    #define IOT_THREAD_MUTEX_NORMAL        0                        /**< Non-robust, deadlock on relock, does not remember owner. */
+#endif
+#ifndef IOT_THREAD_MUTEX_ERRORCHECK
+    #define IOT_THREAD_MUTEX_ERRORCHECK    1                        /**< Non-robust, error on relock,  remembers owner. */
+#endif
+#ifndef IOT_THREAD_MUTEX_RECURSIVE
+    #define IOT_THREAD_MUTEX_RECURSIVE     2                        /**< Non-robust, recursive relock, remembers owner. */
+#endif
+#ifndef IOT_THREAD_MUTEX_DEFAULT
+    #define IOT_THREAD_MUTEX_DEFAULT       IOT_THREAD_MUTEX_NORMAL  /**< IOT_THREAD_MUTEX_NORMAL (default). */
+#endif
+/**@} */
+
+/*-----------------------------------------------------------*/
+
+static void _threadRoutineWrapper( void * pArgument )
+{
+    _threadInfo_t * pThreadInfo = ( _threadInfo_t * ) pArgument;
+
+    /* Run the thread routine. */
+    pThreadInfo->threadRoutine( pThreadInfo->pArgument );
+    IotThreads_Free( pThreadInfo );
+
+    vTaskDelete( NULL );
+}
+
+/*-----------------------------------------------------------*/
+
+bool Iot_CreateDetachedThread(  IotThreadRoutine_t threadRoutine,
+                                void * pArgument,
+                                int32_t priority,
+                                size_t stackSize )
+{
+    IotLogDebug( "Creating new thread." );
+    int status = 0;
+    _threadInfo_t * pThreadInfo = IotThreads_Malloc(sizeof(_threadInfo_t));
+
+    if ( pThreadInfo == NULL)
+    {
+        IotLogDebug( "Unable to allocate memory for threadRoutine %p.", threadRoutine );
+        status = 1;
+    }
+
+    /* Create the FreeRTOS task that will run the thread. */
```

Comment:
FR is careful to not use the word thread, but use Task instead. Should we update the comments here to use Task instead?

Reply:
In the C SDK, I am careful to always say "thread" over "task".  Not sure which is better.

Reply:
Richard writes this in "Mastering the FreeRTOS™ Real Time Kernel"

A Note About Terminology
In FreeRTOS, each thread of execution is called a ‘task’. There is no consensus on
terminology within the embedded community, but I prefer ‘task’ to ‘thread,’ as thread can have
a more specific meaning in some fields of application.